shopfloor/services/zone_picking.py

```diff
@@ -803,10 +835,18 @@ def _set_destination_location(self, move_line, quantity, confirmation, location)
             )
             return (location_changed, response)
         # destination location set to the scanned one
-        self._write_destination_on_lines(move_line, location)
+
         stock = self._actions_for("stock")
-        stock.mark_move_line_as_picked(move_line, quantity)
-        stock.validate_moves(move_line.move_id)
+        move_lines = move_line
+        if package:
+            quantity = None
+            # Handling all move lines from a complete mix pack
+            for move_line in package.move_line_ids:
+                move_line.qty_done = move_line.product_uom_qty
+                move_lines |= move_line
```

Comment:
This will cause an error if the package was already used.
We should check the state or the picking of the move_lines
```suggestion
            for _move_line in package.move_line_ids:
                if _move_line.state != "assigned":
                    continue
                _move_line.qty_done = _move_line.product_uom_qty
                move_lines |= _move_line
```
or 
```suggestion
            for _move_line in package.move_line_ids:
                if _move_line.picking_id != move_line.picking_id:
                    continue
                _move_line.qty_done = _move_line.product_uom_qty
                move_lines |= _move_line
```

Reply:
Updated